packages/framework/src/Model/Store/StoreFactory.php

```diff
@@ -4,14 +4,26 @@
 
 namespace Shopsys\FrameworkBundle\Model\Store;
 
+use Shopsys\FrameworkBundle\Component\EntityExtension\EntityNameResolver;
+
 class StoreFactory
 {
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\EntityExtension\EntityNameResolver $entityNameResolver
+     */
+    public function __construct(
+        protected readonly EntityNameResolver $entityNameResolver,
+    ) {
+    }
+
     /**
      * @param \Shopsys\FrameworkBundle\Model\Store\StoreData $storeData
      * @return \Shopsys\FrameworkBundle\Model\Store\Store
      */
     public function create(StoreData $storeData): Store
     {
-        return new Store($storeData);
+        $classData = $this->entityNameResolver->resolve(Store::class);
```

Comment:
if you think my previous comment makes sense, I would appreciate if you renamed the local variables in all the factories :pray: